### DISCUSSION TITLE
What does safe and dropTable in migrations config actually means?

### DISCUSSION THREAD

**Discussion by artsiommiksiuk** [mikro-orm/mikro-orm#5401] [Q&A] [answered]
So, I'm trying to make safe migrations, meaning no columns drop, and no table drops shall be generated.

In the configs there are safe and dropTable configs for that.

https://mikro-orm.io/docs/migrations#configuration

But the way fields are described I don't understand what does those parameters actually do.
If dropTables: false - migrator still generated drop table statements, if set true it doesn't
With safe: true / false - I didn't seem to get different result.

I'd assume that dropTables: false, and safe: true should do the trick, but I seemed to be not getting this behaviour.

So, the question, how can I achieve such "safe" behaviour from migrator I described?

**Comment by B4nan** [comment, marked as answer]
> If dropTables: false - migrator still generated drop table statements, if set true it doesn't
> With safe: true / false - I didn't seem to get different result.

I don't think it behaves this way. `dropTables` controls whether any `drop table` queries are issued by the schema generator:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/schema/SqlSchemaGenerator.ts#L281

If you enable safe mode, it implies `dropTables = false` (and also blocks `drop column` queries and maybe something else, not 100% sure now).

If it doesn't work for you this way, repro welcome.

**Reply by artsiommiksiuk** [reply]
Okay, you right. Using only safe = true works as you've described. Thanks!
